### DISCUSSION TITLE
How can I access request headers during SSR in the Apollo Client?

### DISCUSSION THREAD

**Discussion by rohit8** [vercel/next.js#15661] [Help]
In the current `with-apollo` example, I'm not able to access request headers during SSR using `setContext` as recommended by Apollo's docs:
```
const authLink = setContext((_, { headers }) => {
 // headers is undefined
});
```
What's the best way to set up apollo client so that request headers exist?

**Comment by sushilbansal** [comment]
Hi @rohit8 , did you find the answer to your question. I am facing the same issue.

**Comment by hanselabreu** [comment]
I believe that `setContext` method only works client side. You need to pass the headers where you're initializing your apollo client.
